### DISCUSSION TITLE
Sequential color scheme for skewed data

### DISCUSSION THREAD

**Discussion by obause** [apache/superset#29472] [Ideas]
## Screenshot
![image](https://github.com/apache/superset/assets/78420842/9a33042a-2994-4e0c-851d-1a8c9288843e)

## Description
I have encountered an issue with the sequential color scheme used in map visualizations when dealing with skewed data. Specifically, when there are only a few values/countries with high metric values, all other values tend to get the same color, making it difficult to distinguish between different lower values. This results in a loss of valuable information and reduces the effectiveness of the visualization.

My idea to solve this would be to use a logarithmic scale for the color mapping. This would ensure a more even distribution of colors across a wide range of values, allowing for better differentiation between lower and higher values.
Also if it would be possible to define specific values at which the color gradient should switch to another color would be helpful. This manual control can help ensure that e.g. the last 5% of data are distinctly visualized vs the first 95%. 

Is one of these two options currently possible? As far as I've seen there is an option for a middle color, but it looks like it is not possible to "move" this middle color e.g. more to the right.

**Comment by rusackas** [comment]
I think it's an interesting idea. For now, besides adding a middle color, you can add MORE colors to the palette and kind of force the skewing that way. 

Regardless, additional controls might offer some flexibility. You're right that we use D3's linear scaling currently. It wouldn't hurt to add an extra control to the relevant charts' control panels to add options here. Our `createLinearScale` method applies the linear scale (obviously) but _could_ be made more generic to support additional scales. A handful of the [many options available](https://d3js.org/d3-scale) might make sense in a select menu. I would just put these in a Select menu, which is hidden by default, and visible when a scalar/gradient color palette is selected. In the name of consistency, I would hope this gets added to all relevant charts, as we're always trying to steer our charts toward having similar capabilities and control panels :)
